src/RequestExecutor/HttpRequestExecutor.spec.ts

```diff
@@ -0,0 +1,176 @@
+import 'reflect-metadata';
+import { HttpRequestExecutor } from './HttpRequestExecutor';
+import {
+  DefaultVirtualScripts,
+  VirtualScript,
+  VirtualScripts
+} from '../Scripts';
+import { Protocol } from './Protocol';
+import { RequestExecutor } from './RequestExecutor';
+import { Request } from './Request';
+import { RequestExecutorOptions } from './RequestExecutorOptions';
+import { expect, should } from 'chai';
+import nock from 'nock';
+import {
+  anyString,
+  anything,
+  instance,
+  mock,
+  reset,
+  verify,
+  when
+} from 'ts-mockito';
+import { container, Lifecycle } from 'tsyringe';
+
+should();
+
+describe('HttpRequestExecutor', () => {
+  const virtualScriptsMock = mock<VirtualScripts>(DefaultVirtualScripts);
+  const virtualScriptMock = mock<VirtualScript>(VirtualScript);
+  const requestMock = mock<Request>(Request);
```

Comment:
The `spy` from `ts-mockito` has limited capabilities in comparison with `mock`. Could you please come up with a working example? I was unable to e.g. assert the number of invocations for a given method.

Reply:
As far as I know, it has the same capabilities. Yet, it is quite buggy, it must preserve an original behavior by definition, unless otherwise noted, but it does not.

```ts
const request = new Request({url: 'https://example.com', headers: {}});
const spiedRequest = spy(request);
when(spiedRequest.method).thenReturn('GET')
const executor = container.resolve<RequestExecutor>(RequestExecutor);

await executor.execute(request);

verify(spiedRequest.setHeaders(testHeaders)).once();
```